registry/config.go

```diff
@@ -23,12 +23,46 @@ type ServiceOptions struct {
 	// V2Only controls access to legacy registries.  If it is set to true via the
 	// command line flag the daemon will not attempt to contact v1 legacy registries
 	V2Only bool `json:"disable-legacy-registry,omitempty"`
+
+	// Registries holds information associated with the specified registries.
+	Registries []Registry `json:"registries,omitempty"`
 }
 
 // serviceConfig holds daemon configuration for the registry service.
 type serviceConfig struct {
 	registrytypes.ServiceConfig
-	V2Only bool
+	V2Only     bool
+	Registries map[string]Registry
+}
+
+// Registry holds information for a registry and its mirrors.
+type Registry struct {
+	// Prefix is used for the lookup of endpoints, where the given registry
+	// is selected when its Prefix is a prefix of the passed reference, for
+	// instance, Prefix:"docker.io/opensuse" will match a `docker pull
+	// opensuse:tumleweed`.
+	Prefix string `json:"Prefix,omitempty"`
+	// The mirrors will be selected prior to the registry during lookup of
+	// endpoints.
+	Mirrors []Mirror `json:"Mirrors,omitempty"`
+	// Host is the host of the registry (e.g., "docker.io")
+	Host string
```

Comment:
How is this different from prefix?

Reply:
It's the actual host, which is required for the endpoint lookup and some sanity checks. I can remove it, but I felt it's cleaner to once parse  *all* relevant when loading the config and then use it throughout the code.

Reply:
Ok, maybe it would be best to expose these kinds of things from method.

Reply:
Sounds good to me. As you said, all data is in the `url.URL`. I'll update the PR and ping you once it's ready for the next iteration.